packages/@ember/-internals/utils/lib/mandatory-setter.ts

```diff
@@ -34,6 +34,12 @@ if (DEBUG) {
 
     SEEN_TAGS!.add(tag);
 
+    if (Array.isArray(obj) && !isNaN(Number(keyName))) {
```

Comment:
After discussion with @krisselden, he suggested we change to using something more like:

```js
function isElementKey(key) {
    return typeof key === 'number' ? isPositiveInt(key) : isStringInt(key);
}

function isStringInt(str) {
    let num = parseInt(str, 10);
    return isPositiveInt(num) && str === '' + num;
}

function isPositiveInt(num) {
    return num >= 0 && num % 1 === 0;
}

// this would replace the conditional
if (isElementKey(keyName)) {
  return;
}
```

Reply:
@pzuraq ping 😺